### PR TITLE
[core] Add "snapshot.expire.clean-empty-directories" to determine whether to clean empty directories when expiring snapshots

### PR DESCRIPTION
**Pull request by zqw142857**
Add a parameter named "data-directories.ignore-clean" to determine whether to ignore the clear data directories.

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 1]
```diff
@@ -982,6 +982,12 @@ public class CoreOptions implements Serializable {
                     .defaultValue(MemorySize.ofMebiBytes(10))
                     .withDescription("The threshold for read file async.");
 
+    public static final ConfigOption<Boolean> DATA_DIRECTORIES_IGNORE_CLEAN =
+            key("data-directories.ignore-clean")
```
Comment (yuzelin) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:986]: Please using "snapshot.expire.ignore-clean-directories"

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -982,6 +982,12 @@ public class CoreOptions implements Serializable {
...
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription("Whether to ignore clean data directories.");
```
Comment (yuzelin) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:989]: 1. Please use key "snapshot.expire.clean-empty-directories" and default value "true".
2. Please make description more clearer, for example, you can use "Whether to try to clean empty directories when expiring snapshots. Note that trying to clean directories might throw exceptions in filesystem, but in most cases it won't cause problems."

**Review by yuzelin — COMMENTED** [review]
And please add a test in `FileDeletionTest` to test that empty directories won't be deleted when setting this option.

**Comment by yuzelin** [comment]
Fixed checkstyle problem.

**Comment by yuzelin** [comment]
Generator doc.

**Review by yuzelin — APPROVED** [review]
+1
